src/identity-providers/routes/EditMapper.ts

```diff
@@ -0,0 +1,27 @@
+import type { LocationDescriptorObject } from "history";
+import { generatePath } from "react-router-dom";
+import type { RouteDef } from "../../route-config";
+import { AddMapper } from "../add/AddMapper";
+
+export type IdentityProviderTab = "mappers";
+
+export type EditIdentityProviderMapperTabParams = {
+  realm: string;
+  id?: string;
+  tab?: IdentityProviderTab;
+  alias?: string;
+  providerId?: string;
+  // editMode: string;
```

Comment:
All parameters here are required in the route, so they should be in this type as well. Also, it looks like some can be removed as they are not present in the route.

```suggestion
  realm: string;
  providerId: string;
  alias: string;
  id: string;
```